infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/LogReplicationClientRouter.java

```diff
@@ -176,6 +176,7 @@ public <T> CompletableFuture<T> sendRequestAndGetCompletable(
             @Nonnull String nodeId) {
 
         HeaderMsg.Builder header = HeaderMsg.newBuilder()
+                .setSession(parameters.getSession())
```

Comment:
No, because the session is used to get the correct sinkManager

Reply:
I meant https://github.com/CorfuDB/CorfuDB/blob/log_replication_v2/infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LogReplicationServer.java#L226 because it also creates a default session.

But thinking a bit further, it will be required if the Source cluster is running an older version.

Reply:
exactly